syntax = "proto2";

import "stream_lite/proto/common.proto";

package job_manager;


message SubmitJobRequest {
  repeated common.Task tasks = 1;
};

message SubmitJobResponse {
  required common.Status status = 1;
  optional string jobid = 2;
}

message RegisterTaskManagerRequest {
  required common.TaskManagerDescription task_manager_desc = 1;
}

message TriggerCheckpointRequest {
  required string jobid = 1;
  optional bool cancel_job = 2 [default = false];
}


service JobManagerService {
  // from user front client
  rpc submitJob(SubmitJobRequest) returns (SubmitJobResponse) {}
  rpc triggerCheckpoint(TriggerCheckpointRequest) returns (common.NilResponse) {}
  
  // from task manager
  rpc registerTaskManager(RegisterTaskManagerRequest) returns (common.NilResponse) {}
};
